### PR TITLE
Simple default replication worker refactor

### PR DESCRIPTION
**Pull request by davinchia**
## What
Logic in this class is going to have to change as part of two big upcoming projects:
* column selection
* progress bars

To prepare for this, I've gone ahead and refactored the `run` method for readability. This is a monster function. The current function is too long and contains several operational abstractions, increasing unnecessary complexity. This is the core of what we do, so it's important to ensure this code is extremely understandable.

Ultimately we want to probably want to break the run method up into two or more separate classes - one that deals with replication and one that deals with outputs - for better testing, readability and isolation. This sets the stage for that.

I have intentionally **NOT** removed or touched any logic, nor have I put thought into consolidating the function signatures to preserve as much of the pre-existing logic and keep the changeset small and reviewable.

This changeset only renames and moves code around.

## How
* Break `run` into `replicate` and `getReplicationOutput`.
* Within `replicate` rename the two runnables to `readFromDstThread` and `readSrcAndWriteDstThread` to further clarify what replicate does.
* Within `getReplicationOutput`, introduce `getTotalStats`, `getPerStreamStats`, `getFailureReasons` and `saveState` to clarify the stages of generating the output.

The `readSrcAndWriteDstThread` function can probably be broken down further. However that is for a separate PR.

## Recommended reading order
1. The one file. I recommend reviewing in an IDE.

## 🚨 User Impact 🚨
Are there any breaking changes? What is the end result perceived by the user? If yes, please merge this PR with the 🚨🚨 emoji so changelog authors can further highlight this if needed.

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>New Connector</strong></summary>

### Community member or Airbyter

- [ ] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the connector is published, connector added to connector index as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</details>

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub and connector version bumped by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</details>

<details><summary><strong>Connector Generator</strong></summary>

- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed

</details>

## Tests

<details><summary><strong>Unit</strong></summary>

*Put your unit tests output here.*

</details>

<details><summary><strong>Integration</strong></summary>

*Put your integration tests output here.*

</details>

<details><summary><strong>Acceptance</strong></summary>

*Put your acceptance tests output here.*

</details>

### CODE REVIEW

File: airbyte-commons-worker/src/main/java/io/airbyte/workers/general/DefaultReplicationWorker.java [hunk 1]
```diff
@@ -139,7 +140,6 @@ public final ReplicationOutput run(final StandardSyncInput syncInput, final Path
     destinationConfig.setCatalog(mapper.mapCatalog(destinationConfig.getCatalog()));
 
     final ThreadedTimeTracker timeTracker = new ThreadedTimeTracker();
-    final long startTime = System.currentTimeMillis();
```
Comment (davinchia) [inline airbyte-commons-worker/src/main/java/io/airbyte/workers/general/DefaultReplicationWorker.java:142]: This is the only deletion in the entire file. We can replace this with `timeTracker.trackReplicationStartTime()` so this is no longer needed.

**Comment by davinchia** [comment]
PTAL @cgardens @jdpgrailsdev @mfsiega-airbyte @benmoriceau @gosusnp 

I'm tagging you all since I've spoken to each and everyone of you on how ugly this function is. Should be a relatively quick review!

**Review by jdpgrailsdev — APPROVED** [review]
:shipit:

File: airbyte-commons-worker/src/main/java/io/airbyte/workers/general/DefaultReplicationWorker.java [hunk 2]
```diff
@@ -427,6 +362,152 @@ private static Runnable getReplicationRunnable(final AirbyteSource source,
...
+      } else {
+        syncStats.setRecordsCommitted(null);
+      }
+      return new StreamSyncStats()
+          .withStreamName(stream)
+          .withStats(syncStats);
+    }).collect(Collectors.toList());
+  }
+
+  private void saveState(StandardSyncInput syncInput, ReplicationOutput output) {
```
Comment (gosusnp) [inline airbyte-commons-worker/src/main/java/io/airbyte/workers/general/DefaultReplicationWorker.java:462]: nit: `saveState` might be a bit misleading here, `output.withState` should be prepping the input for the next activity which should be `PersistStateActivity`.
I don't have a better name, at the moment, and I am hoping to be able to remove this soon, so it's probably not a big deal at the moment.

**Review by gosusnp — APPROVED** [review]

File: airbyte-commons-worker/src/main/java/io/airbyte/workers/general/DefaultReplicationWorker.java (hunk 2 above)
Comment (davinchia) [reply airbyte-commons-worker/src/main/java/io/airbyte/workers/general/DefaultReplicationWorker.java:462]: good point! What about `prepStateForLaterSaving`? I can add a comment that this is done in the PersisStateActivity.

Comment (gosusnp) [reply airbyte-commons-worker/src/main/java/io/airbyte/workers/general/DefaultReplicationWorker.java:462]: Sounds good to me. Name looks a bit awkward but it is probably closer to the truth than saveState.

Comment (davinchia) [reply airbyte-commons-worker/src/main/java/io/airbyte/workers/general/DefaultReplicationWorker.java:462]: accuracy over awkwardness!

Comment (davinchia) [reply airbyte-commons-worker/src/main/java/io/airbyte/workers/general/DefaultReplicationWorker.java]: done!

**Review by benmoriceau — APPROVED** [review]

**Pull request merged**